src/ejabberd_s2s_out.erl

```diff
@@ -329,44 +351,41 @@ wait_for_stream(closed, StateData) ->
 
 
 -spec wait_for_validation(ejabberd:xml_stream_item(), state()) -> fsm_return().
-wait_for_validation({xmlstreamelement, El}, StateData) ->
-    case is_verify_res(El) of
-        {result, To, From, Id, Type} ->
+wait_for_validation({xmlstreamelement, El}, StateData = #state{from_to = FromTo}) ->
+    case mongoose_s2s_dialback:parse_validity(El) of
+        {step_3, FromTo, StreamID, IsValid} ->
+            ?LOG_DEBUG(#{what => s2s_receive_verify,
+                         from_to => FromTo, stream_id => StreamID, is_valid => IsValid}),
+            case StateData#state.verify of
+                false ->
+                    %% TODO: Should'nt we close the connection here ?
```

Comment:
Whenever I see a TODO like this in the code, I am wondering if you have simply forgotten to address it.
If not, I think it's better to state what you know than leave an open question, because IMO it is unlikely to be answered by anyone else.

Reply:
```suggestion
                    %% TODO: Shouldn't we close the connection here ?
```